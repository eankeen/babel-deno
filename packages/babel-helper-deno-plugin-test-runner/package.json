{
  "name": "babel-helper-deno-plugin-test-runner",
  "version": "0.4.0",
  "description": "",
  "main": "build",
  "scripts": {
    "clean": "rimraf build",
    "build": "tsc --build",
    "watch": "tsc --build --watch",
    "test": "jest"
  },
  "keywords": [],
  "author": "Edwin Kofler",
  "license": "MIT",
  "devDependencies": {
    "@babel/core": "^7.9.6",
    "rimraf": "^3.0.2",
    "typescript": "^3.9.2"
  },
  "peerDependencies": {
    "@babel/core": "^7.9.6"
  }
}
