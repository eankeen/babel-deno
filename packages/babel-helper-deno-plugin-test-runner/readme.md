# babel-helper-deno-plugin-test-runner
