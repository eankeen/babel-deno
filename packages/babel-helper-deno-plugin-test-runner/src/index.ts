import path from 'path'
import { getTestFiles, readTranspileAndWrite } from './test.utils'

export function runner(dirname: string) {
  const fixturesDirPath = path.join(dirname, 'fixtures')
  const pluginFilePath = path.join(dirname, '../src/index.ts')
  const pluginName = path
    .basename(path.join(dirname, '../'))
    .slice('babel-plugin-deno-'.length)

  describe(`integration: plugin-${pluginName}`, () => {
    const testFiles = getTestFiles(fixturesDirPath)
    for (const testFile of testFiles) {
      /* eslint-disable-next-line no-loop-func */
      test(`snapshot: ${testFile}`, async () => {
        const { denoResult } = await readTranspileAndWrite(
          pluginFilePath,
          fixturesDirPath,
          testFile
        )

        expect(denoResult.code).toMatchSnapshot()
      })
    }
  })
}
