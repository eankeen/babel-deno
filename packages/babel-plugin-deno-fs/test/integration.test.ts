import { runner } from 'babel-helper-deno-plugin-test-runner';

runner(__dirname);
