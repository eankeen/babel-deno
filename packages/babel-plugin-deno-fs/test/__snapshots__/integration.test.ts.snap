// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration: plugin-fs snapshot: chmodSync 1`] = `
"// invalid
Deno.chmodSync(438);
Deno.chmodSync(path, 438); // correct

Deno.chmodSync(path, 438);
Deno.chmodSync('string', 438); // fs.chmodSync(Buffer.from(), mode)
// fs.chmodSync(new URL(), mode)
// fs.chmodSync('string2', 'modeString')

Deno.chmodSync('string3', 438);"
`;

exports[`integration: plugin-fs snapshot: chownSync 1`] = `
"Deno.chownSync(path, mode);
Deno.chownSync('string', mode); // fs.chownSync(Buffer.from(), mode)
// fs.chownSync(new URL(), mode)
// fs.chownSync('string2', 'modeString')

Deno.chownSync('string3', 0o755);"
`;

exports[`integration: plugin-fs snapshot: closeSync 1`] = `
"// invalid
Deno.closeSync(); //  correct

Deno.closeSync(fd);
Deno.closeSync(2);"
`;

exports[`integration: plugin-fs snapshot: copyFileSync 1`] = `
"Deno.copyFileSync(src, dest);
Deno.copyFileSync(src, dest); // fs.copyFileSync(Buffer.from(), Buffer.from(), mode)
// fs.copyFileSync(new URL(), new URL(), mode)"
`;

exports[`integration: plugin-fs snapshot: existsSync 1`] = `
"import * as denoFs from \\"https://deno.land/x/std@0.53.0/fs/mod.ts\\";
denoFs.existsSync(path); // fs.existsSync(Buffer.from())
// fs.existsSync(new URL())"
`;

exports[`integration: plugin-fs snapshot: mkdirSync 1`] = `
"Deno.mkdirSync(path, options);
Deno.mkdirSync('path', {
  recursive: true
});
Deno.mkdirSync('path2', {
  mode: 'string'
});
Deno.mkdirSync('path3', {
  recursive: true,
  mode: 0o755
});"
`;

exports[`integration: plugin-fs snapshot: mkdtempSync 1`] = `
"fs.mkdtempSync(prefix, options);
fs.mkdtempSync('prefix', {});
fs.mkdtempSync('prefix', {
  encoding: 'utf8'
});"
`;
